src/operation/watch/test.rs

```diff
@@ -0,0 +1,232 @@
+use std::time::Duration;
+
+use crate::{
+    bson::{doc, Document},
+    bson_util,
+    change_stream::ChangeStreamTarget,
+    cmap::{CommandResponse, StreamDescription},
+    operation::{Operation, Watch},
+    options::{
+        ChangeStreamOptions,
+        FullDocumentType,
+        ReadPreference,
+        SelectionCriteria,
+        StreamAddress,
+    },
+    Namespace,
+};
+
+fn build_test(
+    target: impl Into<ChangeStreamTarget>,
+    pipeline: Vec<Document>,
+    options: Option<ChangeStreamOptions>,
+    mut expected_body: Document,
+) {
+    let watch = Watch::new(target.into(), pipeline, options).unwrap();
+
+    let mut cmd = watch.build(&StreamDescription::new_testing()).unwrap();
+
+    assert_eq!(cmd.name.as_str(), "aggregate");
+
+    bson_util::sort_document(&mut expected_body);
+    bson_util::sort_document(&mut cmd.body);
+
+    assert_eq!(cmd.body, expected_body);
+}
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+async fn build() {
+    let pipeline = vec![doc! {"$match": { "x": 3 }}];
+    let options = ChangeStreamOptions::builder()
+        .selection_criteria(SelectionCriteria::ReadPreference(ReadPreference::Primary))
+        .max_await_time(Duration::from_millis(5))
+        .batch_size(10)
+        .full_document(FullDocumentType::UpdateLookup)
+        .build();
+
+    let expected_body = doc! {
+        "aggregate": 1,
+        "cursor": {"batchSize": 10},
+        "pipeline": [
+            {"$changeStream": {"fullDocument": "updateLookup"}},
```

Comment:
re Patrick's comment above, we also tend to add a space after `{` and before `}`, so this should be 
`{ "$changeStream": { "fullDocument": "updateLookup" } }` (and ditto elsewhere)

Reply:
I think its fixed everywhere else now!